posthog/cdp/templates/reddit/template_reddit_pixel.py

```diff
@@ -0,0 +1,236 @@
+from posthog.cdp.templates.hog_function_template import HogFunctionMappingTemplate, HogFunctionTemplate
+
+common_inputs = [
+    {
+        "key": "eventProperties",
+        "type": "dictionary",
+        "description": "Map of Reddit event attributes and their values. Check out these pages for more details: https://business.reddithelp.com/s/article/manual-conversion-events-with-the-reddit-pixel and https://business.reddithelp.com/s/article/about-event-metadata",
+        "label": "Event parameters",
+        "default": {
+            "conversion_id": "{event.uuid}",
+            "products": "{event.properties.products ? arrayMap(product -> ({'id': product.product_id, 'category': product.category, 'name': product.name}), event.properties.products) : event.properties.product_id ? [{'id': event.properties.product_id, 'category': event.properties.category, 'name': event.properties.name}] : undefined}",
+            "value": "{toFloat(event.properties.value ?? event.properties.revenue ?? event.properties.price)}",
+            "currency": "{event.properties.currency}",
+        },
+        "secret": False,
+        "required": False,
+    },
+]
+
+template_reddit_pixel: HogFunctionTemplate = HogFunctionTemplate(
+    free=True,
+    status="alpha",
+    type="site_destination",
+    id="template-reddit-pixel",
+    name="Reddit Pixel",
+    description="Track how many Reddit users interact with your website.",
+    icon_url="/static/services/reddit.png",
```

Comment:
We don't have a reddit.png at the moment. You can create icons from this endpoint: https://us.posthog.com/api/projects/2/hog_functions/icons?query=reddit

Reply:
Just grabbed one from here https://redditbrand.lingoapp.com/s/Logo-d9x3n2